node/utils/pickup.ts

```diff
@@ -0,0 +1,87 @@
+const DAYS_ORDER = [1, 2, 3, 4, 5, 6, 0]
+const MONDAY_INDEX = 0
+const FRIDAY_INDEX = 4
+
+const normalizeBusinessHours = (businessHours: BusinessHour[]) => {
+  return DAYS_ORDER.map(dayNumber => {
+    const day = businessHours.find(bh => dayNumber === bh.DayOfWeek)
+
+    const dayObject = day && {
+      closed: false,
+      openingTime: day.OpeningTime,
+      closingTime: day.ClosingTime,
+    }
+
+    return {
+      dayNumber,
+      closed: true,
+      openingTime: '',
+      closingTime: '',
+      ...(dayObject ?? {}),
+    }
+  })
+}
+
+const doesWeekDaysHaveTheSameHours = (
+  businessHours: NormalizedBusinessHour[]
+) => {
+  const weekDays = businessHours.slice(MONDAY_INDEX, FRIDAY_INDEX)
+
+  let previousOpeningTime = weekDays[0].openingTime
+  let previousClosingTime = weekDays[0].closingTime
+
+  for (let i = 1; i < weekDays.length; i++) {
+    const currentDay = weekDays[i]
+
+    if (
+      currentDay.openingTime !== previousOpeningTime ||
+      currentDay.closingTime !== previousClosingTime
+    ) {
+      return false
+    }
+
+    previousOpeningTime = currentDay.openingTime
+    previousClosingTime = currentDay.closingTime
```

Comment:
I think those lines are unnecessary, as `previousOpeningTime` will always be equal to `currentDay.openingTime` (same goes for closing time).

But if you make this change, maybe changing `previousTime` to `firstTime` or something like that, will make more sense.

Reply:
Nice catch! Done!